crates/revm/src/state_change.rs

```diff
@@ -158,3 +167,95 @@ pub fn insert_post_block_withdrawals_balance_increments(
         }
     }
 }
+
+/// Applies the post-block call to the EIP-7002 withdrawal requests contract.
+///
+/// If Prague is not active at the given timestamp, then this is a no-op, and an empty vector is
+/// returned. Otherwise, the withdrawal requests are returned.
+#[inline]
+pub fn apply_withdrawal_requests_contract_call<EXT, DB: Database + DatabaseCommit>(
+    chain_spec: &ChainSpec,
+    block_timestamp: u64,
+    evm: &mut Evm<'_, EXT, DB>,
+) -> Result<Vec<Request>, BlockExecutionError>
+where
+    DB::Error: std::fmt::Display,
+{
+    if !chain_spec.is_prague_active_at_timestamp(block_timestamp) {
+        return Ok(vec![])
+    }
```

Comment:
we're checking this condition always twice

imo this should not be part of this call and the caller must decide when to invoke this